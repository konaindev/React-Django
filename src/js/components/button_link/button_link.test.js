import React from "react";
import renderer from "react-test-renderer";

import ButtonLink from "./index";

describe("ButtonLink", () => {
  it("renders link correctly", () => {
    const tree = renderer
      .create(<ButtonLink link="http://www.test.com/" target="_blank" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
