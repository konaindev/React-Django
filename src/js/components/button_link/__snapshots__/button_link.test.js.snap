// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders link correctly 1`] = `
<a
  className="button-link"
  href="http://www.test.com/"
  target="_blank"
>
  <svg
    aria-hidden="true"
    className="button-link__icon"
    color="inherit"
    fill="none"
    focusable="false"
    height="1em"
    role="presentation"
    viewBox="0 0 20 20"
    width="1em"
  >
    <path
      d="M9.99935 18.3327C14.6017 18.3327 18.3327 14.6017 18.3327 9.99935C18.3327 5.39698 14.6017 1.66602 9.99935 1.66602C5.39698 1.66602 1.66602 5.39698 1.66602 9.99935C1.66602 14.6017 5.39698 18.3327 9.99935 18.3327Z"
      stroke="#798796"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.5"
    />
    <path
      d="M1.66602 10H18.3327"
      stroke="#798796"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.5"
    />
    <path
      d="M9.99935 1.66602C12.0837 3.94798 13.2683 6.90938 13.3327 9.99935C13.2683 13.0893 12.0837 16.0507 9.99935 18.3327C7.91495 16.0507 6.73039 13.0893 6.66602 9.99935C6.73039 6.90938 7.91495 3.94798 9.99935 1.66602V1.66602Z"
      stroke="#798796"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.5"
    />
  </svg>
  test.com
</a>
`;
